protocol/storage/client.go

```diff
@@ -225,10 +225,6 @@ func (smc *Client) recordResponse(resp *storagedeal.Response, miner address.Addr
 	if !proposalCid.Equals(resp.ProposalCid) {
 		return fmt.Errorf("cids not equal %s %s", proposalCid, resp.ProposalCid)
 	}
-	storageDeal := smc.api.DealGet(proposalCid)
-	if storageDeal != nil {
-		return fmt.Errorf("deal [%s] is already in progress", proposalCid.String())
-	}
```

Comment:
This removal looks like an unintended semantic change: if there is *no* error from DealGet, this should raise one. OTOH, if the error is "not found", this needs to continue.